taf/updater/updater_pipeline.py

```diff
@@ -1329,19 +1315,32 @@ def _merge_commit(
             f"Please update the repository at {repository.path} manually and try again."
         )
 
-    commit_merged = False
-    if force_revert:
-        # check if repository already contains this commit that needs to be merged
-        # and commits following it
-        commits_since_last_validated = repository.all_commits_since_commit(
-            commit_to_merge
+    if repository.top_commit_of_branch(branch) == commit_to_merge:
+        return
+    commits_since_to_merge = repository.all_commits_since_commit(
+        commit_to_merge, branch=branch
+    )
+    if not len(commits_since_to_merge):
+        taf_logger.info(
+            "{} Merging commit {} into branch {}",
+            repository.name,
+            format_commit(commit_to_merge),
+            branch,
         )
-        if len(commits_since_last_validated):
-            repository.reset_to_commit(commit_to_merge, hard=True)
-            commit_merged = True
-
-    if not commit_merged:
         repository.merge_commit(commit_to_merge)
-    if checkout:
-        taf_logger.info("{}: checking out branch {}", repository.name, branch)
-        repository.checkout_branch(branch)
+
+    elif len(commits_since_to_merge) and force_revert:
+        taf_logger.info(
+            "{} Reverting branch {} to {}",
+            repository.name,
+            branch,
+            format_commit(commit_to_merge),
+        )
+        repository.reset_to_commit(commit_to_merge, hard=True)
```

Comment:
Is the reason why we don't merge the `commit_to_merge` n this condition because merging a newer commit will lead to fast forward? If that's the case, might want to write that down somewhere in here, since it seems confusing as to why we're resetting to commit?